tidb log print "connection running loop panic  [err="Trying to start aggressive locking while it's already started"] " when run tpcc and import into

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1、run tpcc and import into
[tidb.tar.gz](https://github.com/pingcap/tidb/files/15431000/tidb.tar.gz)

### 2. What did you expect to see? (Required)
no panic

### 3. What did you see instead (Required)
tidb log print “connection running loop panic

```
[2024/05/24 10:49:13.008 +08:00] [ERROR] [conn.go:1013] ["connection running loop panic"] [conn=2594177614] [session_alias=] [lastSQL=ROLLBACK] [err="Trying to start aggressive locking while it's already started"] [stack="github.com/pingcap/tidb/pkg/server.(*clientConn).Run.func1
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:1016
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:914
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).Exec.func1
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/adapter.go:452
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:914
github.com/tikv/client-go/v2/txnkv/transaction.(*KVTxn).StartAggressiveLocking
	/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20240516031922-38e0dca30c8c/txnkv/transaction/txn.go:919
github.com/pingcap/tidb/pkg/store/driver/txn.(*tikvTxn).StartFairLocking
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/store/driver/txn/txn_driver.go:422
github.com/pingcap/tidb/pkg/session.(*LazyTxn).StartFairLocking
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/txn.go:495
github.com/pingcap/tidb/pkg/sessiontxn/isolation.(*basePessimisticTxnContextProvider).OnPessimisticStmtStart
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/sessiontxn/isolation/base.go:683
github.com/pingcap/tidb/pkg/session.(*txnManager).OnPessimisticStmtStart
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/txnmanager.go:228
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).handlePessimisticDML
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/adapter.go:998
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).handleNoDelay
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/adapter.go:791
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).Exec
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/adapter.go:559
github.com/pingcap/tidb/pkg/session.runStmt
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/session.go:2285
github.com/pingcap/tidb/pkg/session.(*session).ExecuteStmt
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/session.go:2146
github.com/pingcap/tidb/pkg/server.(*TiDBContext).ExecuteStmt
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/driver_tidb.go:294
github.com/pingcap/tidb/pkg/server.(*clientConn).handleStmt
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:2021
github.com/pingcap/tidb/pkg/server.(*clientConn).handleQuery
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:1774
github.com/pingcap/tidb/pkg/server.(*clientConn).dispatch
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:1348
github.com/pingcap/tidb/pkg/server.(*clientConn).Run
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:1114
github.com/pingcap/tidb/pkg/server.(*Server).onConn
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/server.go:739"]
[2024/05/24 10:49:13.008 +08:00] [ERROR] [terror.go:324] ["encountered error"] [error="trying to rollback transaction when aggressive locking is pending"] [stack="github.com/pingcap/tidb/pkg/parser/terror.Log
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/parser/terror/terror.go:324
github.com/pingcap/tidb/pkg/session.(*session).RollbackTxn
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/session.go:941
github.com/pingcap/tidb/pkg/session.(*session).Close
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/session.go:2516
github.com/pingcap/tidb/pkg/server.(*TiDBContext).Close
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/driver_tidb.go:313
github.com/pingcap/tidb/pkg/server.closeConn.func1
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:400
sync.(*Once).doSlow
	/usr/local/go/src/sync/once.go:74
sync.(*Once).Do
	/usr/local/go/src/sync/once.go:65
github.com/pingcap/tidb/pkg/server.closeConn
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:380
github.com/pingcap/tidb/pkg/server.(*clientConn).Close
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:373
github.com/pingcap/tidb/pkg/server.(*clientConn).Run.func1
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:1023
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:914
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).Exec.func1
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/adapter.go:452
runtime.gopanic
	/usr/local/go/src/runtime/panic.go:914
github.com/tikv/client-go/v2/txnkv/transaction.(*KVTxn).StartAggressiveLocking
	/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20240516031922-38e0dca30c8c/txnkv/transaction/txn.go:919
github.com/pingcap/tidb/pkg/store/driver/txn.(*tikvTxn).StartFairLocking
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/store/driver/txn/txn_driver.go:422
github.com/pingcap/tidb/pkg/session.(*LazyTxn).StartFairLocking
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/txn.go:495
github.com/pingcap/tidb/pkg/sessiontxn/isolation.(*basePessimisticTxnContextProvider).OnPessimisticStmtStart
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/sessiontxn/isolation/base.go:683
github.com/pingcap/tidb/pkg/session.(*txnManager).OnPessimisticStmtStart
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/txnmanager.go:228
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).handlePessimisticDML
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/adapter.go:998
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).handleNoDelay
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/adapter.go:791
github.com/pingcap/tidb/pkg/executor.(*ExecStmt).Exec
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/executor/adapter.go:559
github.com/pingcap/tidb/pkg/session.runStmt
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/session.go:2285
github.com/pingcap/tidb/pkg/session.(*session).ExecuteStmt
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/session/session.go:2146
github.com/pingcap/tidb/pkg/server.(*TiDBContext).ExecuteStmt
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/driver_tidb.go:294
github.com/pingcap/tidb/pkg/server.(*clientConn).handleStmt
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:2021
github.com/pingcap/tidb/pkg/server.(*clientConn).handleQuery
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:1774
github.com/pingcap/tidb/pkg/server.(*clientConn).dispatch
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:1348
github.com/pingcap/tidb/pkg/server.(*clientConn).Run
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/conn.go:1114
github.com/pingcap/tidb/pkg/server.(*Server).onConn
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/tidb/pkg/server/server.go:739"]`
```

### 4. What is your TiDB version? (Required)
./tidb-server -V
 Release Version: v8.2.0-alpha
Edition: Community
Git Commit Hash: 3485857563aaeb040c0eab2057ce4a70dd3b6e2f
Git Branch: heads/refs/tags/v8.2.0-alpha
UTC Build Time: 2024-05-23 11:45:56
GoVersion: go1.21.10
Race Enabled: false

--- a/DEPS.bzl
+++ b/DEPS.bzl
@@ -7041,13 +7041,13 @@ def go_deps():
         name = "com_github_tikv_client_go_v2",
         build_file_proto_mode = "disable_global",
         importpath = "github.com/tikv/client-go/v2",
-        sha256 = "48097286e81b799d8cc8612879e4d3fa569be79a1be93ed77131c04052c3a069",
-        strip_prefix = "github.com/tikv/client-go/v2@v2.0.8-0.20240531122021-7a74511a5241",
+        sha256 = "24e9d31e6a51a3f40274a541b6ef93af8172d93e5ebf5ee5a66692473dd1a98a",
+        strip_prefix = "github.com/tikv/client-go/v2@v2.0.8-0.20240716121331-2cd3a741ea8e",
         urls = [
-            "http://bazel-cache.pingcap.net:8080/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240531122021-7a74511a5241.zip",
-            "http://ats.apps.svc/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240531122021-7a74511a5241.zip",
-            "https://cache.hawkingrei.com/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240531122021-7a74511a5241.zip",
-            "https://storage.googleapis.com/pingcapmirror/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240531122021-7a74511a5241.zip",
+            "http://bazel-cache.pingcap.net:8080/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240716121331-2cd3a741ea8e.zip",
+            "http://ats.apps.svc/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240716121331-2cd3a741ea8e.zip",
+            "https://cache.hawkingrei.com/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240716121331-2cd3a741ea8e.zip",
+            "https://storage.googleapis.com/pingcapmirror/gomod/github.com/tikv/client-go/v2/com_github_tikv_client_go_v2-v2.0.8-0.20240716121331-2cd3a741ea8e.zip",
         ],
     )
     go_repository(

--- a/go.mod
+++ b/go.mod
@@ -104,7 +104,7 @@ require (
 	github.com/stretchr/testify v1.8.4
 	github.com/tdakkota/asciicheck v0.2.0
 	github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2
-	github.com/tikv/client-go/v2 v2.0.8-0.20240531122021-7a74511a5241
+	github.com/tikv/client-go/v2 v2.0.8-0.20240716121331-2cd3a741ea8e
 	github.com/tikv/pd/client v0.0.0-20240708075403-19f65c59bb89
 	github.com/timakin/bodyclose v0.0.0-20230421092635-574207250966
 	github.com/twmb/murmur3 v1.1.6

